### PR TITLE
DOCS-4659 lib injection for hosts and containers

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
documents how to inject tracing libraries into Agents and apps on hosts and containers

### Motivation
DOCS-4658

### Preview

https://docs-staging.datadoghq.com/kari/docs-4659-lib-injection/tracing/trace_collection/library_injection

### Additional Notes

Everything on the "Kubernetes" tab is just brought forward from the previous admission_controller page, not new content

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/library_injection.md [hunk 1]
```diff
@@ -0,0 +1,749 @@
...
+   curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
+   sudo apt install nodejs -y
+   ```
+   For .NET applications, ensure you have the [.NET runtime installed][3].
+
+2. If you haven't already, install your app.
+
+## Configure the injection
+
+The following environment variables configure library injection. You can pass these in by `export` on the command line (`export DD_CONFIG_SOURCES=BASIC`), in your shell configuration, or by passing them in on the launch command:
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:234]:
```suggestion
The following environment variables configure library injection. You can pass these in by `export` through the command line (`export DD_CONFIG_SOURCES=BASIC`), shell configuration, or launch command:
```

File: content/en/tracing/trace_collection/library_injection.md [hunk 2]
```diff
@@ -0,0 +1,749 @@
...
+DD_CONFIG_SOURCES=LOCAL:/etc/<SERVICE_2>/config.yaml;BASIC node index.js &
+```
+
+**.NET app example**:
+```sh
+DD_CONFIG_SOURCES=BASIC dotnet <SERVICE_1>.dll &
+DD_CONFIG_SOURCES=LOCAL:/etc/<SERVICE_2>/config.yaml;BASIC dotnet <SERVICE_2>.dll &
+```
+
+Exercise your application to start generating telemetry data, which you can see as [traces in APM][4].
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:357]:
```suggestion
Execute your application to start generating telemetry data, which you can see as [traces in APM][4].
```

assuming "execute" is meant? unless exercising an application is also a thing

File: content/en/tracing/trace_collection/library_injection.md [hunk 3]
```diff
@@ -0,0 +1,749 @@
...
+[3]: https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu
+[4]: https://app.datadoghq.com/apm/traces
+{{% /tab %}}
+
+{{% tab "Agent on host, app in containers" %}}
+
+<div class="alert alert-info">Tracing library injection on hosts and containers is in beta.</a></div>
+
+
+When your Agent is running on a host your services are running in containers, Datadog injects the tracing library by using a library that is preloaded and that overrides calls to `execve`, and a `runc` shim that intercepts container creation, and configures the initial process launched in a Docker container.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:370]:
```suggestion
When your Agent is running on a host, and your services are running in containers, Datadog injects the tracing library by using a preload library that overrides calls to `execve`. Datadog also uses a `runc` shim that intercepts container creation and configures the initial process launched in a Docker container.
```

File: content/en/tracing/trace_collection/library_injection.md [hunk 4]
```diff
@@ -0,0 +1,749 @@
...
+1. Run the following commands, for example, with `apt`, where `<LANG>` is one of `java`, `nodejs`, `dotnet`, or `all`:
+
+   ```sh
+   sudo apt update
+   sudo apt install -y datadog-apm-inject datadog-apm-library-<LANG>
+   dd-install-ld-preload
+   dd-install-docker-shim
+   ```
+
+2. The preload library injection only works in a new shell, so exit and open a new shell.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:392]:
```suggestion
2. Exit and open a new shell to use the preload library.
```

File: content/en/tracing/trace_collection/library_injection.md [hunk 5]
```diff
@@ -0,0 +1,749 @@
...
+
+If the environment variables `DD_ENV`, `DD_SERVICE`, or `DD_VERSION` are specified in a service container image, those values are used to tag telemetry from the container.
+
+If they are not specified, `DD_ENV` uses the `env` value set in the `/etc/datadog-agent/inject/docker_config.yaml` config file, if any. `DD_SERVICE` and `DD_VERSION` are derived from the name of the Docker image. An image with the name `my-service:1.0` is tagged with `DD_SERVICE` of `my-service` and a `DD_VERSION` of `1.0`.
+
+## Launch your services
+
+Start your Agent and launch your containerized services as usual.
+
+Exercise your application to start generating telemetry data, which you can see as [traces in APM][4].
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:522]:
```suggestion
Execute your application to start generating telemetry data, which you can see as [traces in APM][4].
```

see previous note

File: content/en/tracing/trace_collection/library_injection.md [hunk 6]
```diff
@@ -0,0 +1,749 @@
...
+[3]: /tracing/trace_collection/library_config/
+[4]: https://app.datadoghq.com/apm/traces
+
+{{% /tab %}}
+
+{{% tab "Containers" %}}
+
+<div class="alert alert-info">Tracing library injection in containers is in beta.</a></div>
+
+When your Agent and services are running in Docker containers on the same host, Datadog injects the tracing library by using a library that is preloaded and that overrides calls to `execve`, and a `runc` shim that intercepts container creation, and configures the initial process launched in a Docker container.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:536]:
```suggestion
When your Agent and services are running in Docker containers on the same host, Datadog injects the tracing library by using a preload library that overrides calls to `execve`. Datadog also uses a `runc` shim that intercepts container creation and configures the initial process launched in a Docker container.
```

File: content/en/tracing/trace_collection/library_injection.md [hunk 7]
```diff
@@ -0,0 +1,749 @@
...
+1. Run the following commands, for example, with `apt`, where `<LANG>` is one of `java`, `nodejs`, `dotnet`, or `all`:
+
+   ```sh
+   sudo apt update
+   sudo apt install -y datadog-apm-inject datadog-apm-library-<LANG>
+   dd-install-ld-preload
+   dd-install-docker-shim
+   ```
+
+2. The preload library injection only works in a new shell, so exit and open a new shell.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:557]:
```suggestion
2. Exit and open a new shell to use the preload library.
```

File: content/en/tracing/trace_collection/library_injection.md [hunk 8]
```diff
@@ -0,0 +1,749 @@
...
+
+```sh
+docker-compose up -d dd-agent
+```
+
+## Launch your services
+
+Launch your containerized services as usual.
+
+Exercise your application to start generating telemetry data, which you can see as [traces in APM][4].
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_injection.md:730]:
```suggestion
Execute your application to start generating telemetry data, which you can see as [traces in APM][4].
```

see previous notes

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/tracing/trace_collection/library_injection.md (hunk 2 above)
Comment (kayayarai) [reply content/en/tracing/trace_collection/library_injection.md:357]: @cswatt Yeah, exercise means "not just run it but also do some stuff in it that causes traces to happen."

File: content/en/tracing/trace_collection/library_injection.md [hunk 9]
```diff
@@ -0,0 +1,749 @@
...
+
+{{% tab "Containers" %}}
+
+<div class="alert alert-info">Tracing library injection in containers is in beta.</a></div>
+
+When your Agent and services are running in Docker containers on the same host, Datadog injects the tracing library by using a preload library that overrides calls to `execve`. Datadog also uses a `runc` shim that intercepts container creation and configures the initial process launched in a Docker container.
+
+Any newly started processes are intercepted and the specified instrumentation library is injected into the services.
+
+# Requirements
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/library_injection.md:540]:
```suggestion
## Requirements
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**